### DISCUSSION TITLE
Codicons spinning off-center in custom tree view

### DISCUSSION THREAD

**Discussion by SanjulaGanepola** [eclipse-theia/theia#11270] [Q&A] [answered]
When trying to apply a spinning animation to my icon as supported in VS Code [here](https://code.visualstudio.com/api/references/icons-in-labels#animation), I notice that in Theia the icons are spinning off-center when used in a custom tree view. I am currently running in Theia v1.18.0 where I noticed the issue, but I also verified the issue in Theia Blueprint v1.26.0.

I see that this was previously a known issue in VS Code (https://github.com/microsoft/vscode/issues/117859), but has since been resolved back in May 2021. Is there any current workaround to support spinning icons or is there a plan to resolve this issue in Theia?

**Comment by vince-fugnitto** [comment, marked as answer]
> Is there any current workaround to support spinning icons or is there a plan to resolve this issue in Theia?

@SanjulaGanepola thank you for the discussion and reporting the issue. I believe we can apply a similar fix as VS Code so that icons do not spin off-center. Are you interested in contributing a fix?

**Reply by SanjulaGanepola** [reply]
Thank you for the reply. I am not very familiar with the current codebase, but I can try to take a look. For now I have gone ahead and opened an issue to track this (#11273).
